refactor documentation (#10)
ROCm-Developer-Tools/rocr_debug_agent: docs/conceptual/user-guide.rst

.. meta::
   :description: A library that can be loaded by ROCr to print the AMDGPU wavefront states

>    :keywords: ROCdebug-agent, ROCm, library, tool, rocr

Reviewer: ```suggestion
   :keywords: ROCdebug-agent, ROCm, library, tool, rocr, ROCm Debug Agent
```

.. _user-guide:

===========================

> ROCdebug-agent user guide

Reviewer: ```suggestion
ROCm Debug Agent user guide
```

Reviewer: We should try to use the proper name as noted on the What is ROCm page. 

Fix it here, too
![image](https://github.com/ROCm/rocr_debug_agent/assets/164940351/abe192d2-bc89-4ff7-8f0c-f9b8858cb31c)

Author: Done

===========================

> To display the source text location with the machine code instructions around the wavefront's pc, compile the AMDGPU code objects with ``-ggdb``.  In addition, you can optionally user ``-O0`` to achieve a more intuitive display of the source text location
> as higher optimization levels can help to reorder machine code instructions. When ``-ggdb`` is not used, the source line information is not available and only machine code instructions starting at the
> wavefront's pc are printed.

Reviewer: ```suggestion
To display the source text location with the machine code instructions around the wavefront's pc, compile the AMDGPU code objects with ``-ggdb``.  In addition, you can optionally use ``-O0`` to achieve a more intuitive display of the source text location, as higher optimization levels can help to reorder machine code instructions. When ``-ggdb`` is not used, the source line information is unavailable, and only machine code instructions starting at the
wavefront's pc are printed.
```

.. code:: shell

    /opt/rocm/bin/hipcc -O0 -ggdb -o my_program my_program.cpp

To use the ROCdebug-agent, set the ``HSA_TOOLS_LIB`` environment variable to the file name or path of the library and the ``HSA_ENABLE_DEBUG`` environment variable to ``1``.

.. code:: shell

    HSA_TOOLS_LIB=/opt/rocm/lib/librocm-debug-agent.so.2 HSA_ENABLE_DEBUG=1 ./my_program

If the application encounters a triggering event, ROCdebug-agent prints the state of some or all AMDGPU wavefronts.

> See a sample print out:

Reviewer: ```suggestion
See a sample printout:
```

        0x01e0: 22222222 11111111 22222222 11111111 22222222 11111111 22222222 11111111

    Disassembly for function vector_add_assert_trap(int*, int*, int*):
        code object: file:////rocm-debug-agent/build/test/rocm-debug-agent-test#offset=14309&size=31336
        loaded at: [0x7fd4f100c000-0x7fd4f100e070]

    /rocm-debug-agent/test/vector_add_assert_trap.cpp:
    55        c[gid] = a[gid] + b[gid] + (lds_check[0] >> 32);
        0x7fd4f100d0c4 <+196>:    s_waitcnt vmcnt(0) lgkmcnt(0)
        0x7fd4f100d0c8 <+200>:    v_add3_u32 v2, v2, v4, v3
        0x7fd4f100d0d0 <+208>:    global_store_dword v[0:1], v2, off
        0x7fd4f100d0d8 <+216>:    s_or_saveexec_b64 s[0:1], s[0:1]
        0x7fd4f100d0dc <+220>:    s_xor_b64 exec, exec, s[0:1]
        0x7fd4f100d0e0 <+224>:    s_cbranch_execz 65503  # 0x7fd4f100d060 <vector_add_assert_trap(int*, int*, int*)+96>

    53          __builtin_trap ();
        0x7fd4f100d0e4 <+228>:    s_mov_b64 s[0:1], s[6:7]
    =>  0x7fd4f100d0e8 <+232>:    s_trap 2
        0x7fd4f100d0ec <+236>:    s_endpgm

    End of disassembly.
    Aborted (core dumped)

The supported triggering events are:

- **Memory fault**

> A memory fault occurs when an AMDGPU accesses an inaccessible page. This leads to printing information about the memory fault as shown below:

Reviewer: ```suggestion
A memory fault occurs when an AMDGPU accesses an inaccessible page. This leads to printing information about the memory fault, as shown below:
```
